framework/model/metric/config.go

```diff
@@ -44,7 +44,8 @@ type AccessLogSourceConfig struct {
 }
 
 type AccessLogConvertorConfig struct {
-	Name      string // handler name
-	ClientSet *kubernetes.Clientset
-	Handler   func(logEntry []*data_accesslog.HTTPAccessLogEntry, clientSet *kubernetes.Clientset) (map[string]map[string]string, error)
+	Name          string // handler name
+	SvcToIpsCache map[string][]string
+	CacheLock     *sync.RWMutex
+	Handler       func(logEntry []*data_accesslog.HTTPAccessLogEntry, svcToIpCache map[string][]string, cacheLock *sync.RWMutex) (map[string]map[string]string, error)
```

Comment:
too strange. this handler is also sth like a interface. I can't image a interface delcaration with a specific lock arg.

Reply:
```go
Handler func(logEntry []*data_accesslog.HTTPAccessLogEntry) (map[string]map[string]string, error)
```

is enough